dashboard/src/shared/schema.ts

```diff
@@ -1,17 +1,110 @@
-  // TBD
-  return {
-    username: {
-      name: "username",
-      path: "wordpressUsername",
-      value: "user",
-    } as IBasicFormParam,
-  };
+// Avoid to explicitly add "null" when an element is not defined
+import { nullOptions } from "yaml/types";
+nullOptions.nullStr = "";
+
+// retrieveBasicFormParams iterates over a JSON Schema properties looking for `form` keys
+// It uses the raw yaml to setup default values.
+// It returns a key:value map for easier handling.
+export function retrieveBasicFormParams(
+  defaultValues: string,
+  schema: any,
+): { [key: string]: IBasicFormParam } {
+  const basicFormParameters = {};
+  if (schema && schema.properties) {
+    const params = lookForFormParams(defaultValues, schema.properties);
+    if (params) {
+      params.forEach(p => (basicFormParameters[p.name] = p));
+    }
+  }
+  return basicFormParameters;
+}
+
+// getDocumentElem returns the YAML element given a root element
+export function getDocumentElem(rootElem: any, path: string): any | undefined {
+  // Split the path per keys
+  const splittedPath = path.split(".");
+  if (splittedPath.length === 1) {
+    // If the path doesn't contain inner objects, simply return
+    return rootElem.get(path);
+  }
+  // If the path contains inner elements, iterate until finding the last one
+  let res = rootElem.get(splittedPath[0]);
+  splittedPath.slice(1, splittedPath.length).forEach(p => (res = res && res.get(p)));
+  return res;
 }
 
+// setValue modifies the current values (text) based on a path
 export function setValue(values: string, path: string, newValue: any) {
-  // TBD
-  return values;
+  const doc = YAML.parseDocument(values);
+  if (path.includes(".")) {
+    // If the path points to an inner element we need to parse it
+    // and obtain the previous element to set it
+    // e.g.
+    // path = "a.b.c"
+    // itemToSet = "c"
+    // elem = "a.b"
+    // elem.set(itemToSet, value)
+    const splittedPath = path.split(".");
+    const itemToSet = splittedPath[splittedPath.length - 1];
+    splittedPath.pop();
+    const pathToSet = splittedPath.join(".");
+    const elem = getDocumentElem(doc, pathToSet);
+    if (!elem) {
+      // TODO: It should be possible to handle this situation and add a new object
+      throw new Error(`Unable to set the value ${path}. Parent element not found`);
+    }
+    elem.set(itemToSet, newValue);
+  } else {
+    doc.set(path, newValue);
+  }
+  return doc.toString();
+}
+
+// getDefaultValues returns the current value of an object based on YAML text and its path
+export function getDefaultValue(defaultValues: string, path: string): string | undefined {
+  const doc = YAML.parseDocument(defaultValues);
+  if (path.includes(".")) {
+    const elem = getDocumentElem(doc, path);
+    // If the path is found, return it as a string
+    return elem && elem.toString();
+  }
+  return path.includes(".") ? getDocumentElem(doc, path).toString() : doc.get(path);
```

Comment:
This condition looks identical to the one above - with an added else? Perhaps a refactory snafu? Also, is `doc.get(path)` a string? Seems you're wanting to return a string above but don't do so in the else clause here (though as above, not clear to me why we'd do that in this function rather than explicitly at the call-site).

Reply:
ups. yes, I added the `if` above and forgot to remove this.